Added Caching of npm packages and added npm install

ohbus opened the pull request:
- [x] Added `npm install` for installing node packages
- [x] Added naming for different actions
- [x] Added caching support of `node_modules` to not download packages for every build.

> Let me know what needs to be changed. Thanks

ohbus committed:
Update node.js.yml

ohbus committed:
Update node.js.yml

ohbus committed:
Update node.js.yml

ohbus committed:
Update node.js.yml

championswimmer commented on .github/workflows/node.js.yml:
```diff
@@ -1,7 +1,7 @@
 # For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
 
-name: Node.js CI
+name: Sclr.ac CI
```
Node.js CI was a better name it defines what this action is, not what this project is 😅

championswimmer commented on .github/workflows/node.js.yml:
```diff
@@ -12,18 +12,40 @@ on:
       matrix:
         node-version: [14.x, 16.x]
         # See supported Node.js release schedule at https://nodejs.org/en/about/releases/
+        os: [ubuntu-latest, macOS-latest, windows-latest]
```
this will eat into our free github action minutes. just ubuntu is fine.

ohbus replied:
As you mentioned it is free, so why not use it 😄

championswimmer replied:
Only 2000 minutes are free. Not unlimited :( 

Also we should not compute just for fun. There is a carbon footprint and environment impact of such small steps. Do we have any potential for cross-OS issues right now? If not why waste computation resources.

ohbus replied:
I thought 2000 minutes are free for private repo.

But I agree with your point on carbon footprint but unless we see any raised issue about the same let us stick with ubuntu

ohbus committed:
updated name

championswimmer commented on .github/workflows/node.js.yml:
```diff
@@ -12,18 +12,40 @@ on:
-    - run: npm run build
+    
+    - name: Install Node packages
+      run: npm install
```
npm ci should run npm install right ?

ohbus replied:
I wouldn't know, I used to follow `npm install` to install the node_modules.

Please tell me what to do?

championswimmer replied:
Use NPM CI 

It is used for installing on CI machines 
https://www.geeksforgeeks.org/difference-between-npm-i-and-npm-ci-in-node-js/

ohbus replied:
Thanks for the information

ohbus committed:
Less Carbonated

Made the environment more carbon-friendly by not using different types of containers.

ohbus committed:
Use npm ci
